test/unit/test-atomic.cpp

```diff
@@ -68,6 +72,10 @@ void testAtomicFunctionBasic()
   EXPECT_EQ((T)4, dest[5]);
   EXPECT_EQ((T)1, dest[6]);
   EXPECT_EQ((T)13, dest[7]);
+  EXPECT_LE((T)0, dest[8]);
+  EXPECT_GT((T)N, dest[8]);
+  EXPECT_LT((T)0, dest[9]);
+  EXPECT_GE((T)N, dest[9]);
```

Comment:
If you have a moment, would you mind also mirroring these changes to the HIP version of the test (testAtomicFunctionBasic_gpu() lines 263-302)? Looks great, and thanks again!

Reply:
that's a good idea.